custom_components/localtuya/cover.py

```diff
@@ -186,19 +220,194 @@ def __init__(
         self,
         device,
         config_entry,
-        switchid,
+        coverid,
         **kwargs,
     ):
-        #_LOGGER.info("running def __init__ of LocaltuyaCover(CoverEntity) with self=%s device=%s name=%s friendly_name=%s icon=%s switchid=%s open_cmd=%s close_cmd=%s stop_cmd=%s", self, device, name, friendly_name, icon, switchid, open_cmd, close_cmd, stop_cmd)
-        super().__init__(device, config_entry, switchid, **kwargs)
+        #_LOGGER.info("running def __init__ of LocaltuyaCover(CoverEntity) with self=%s device=%s name=%s friendly_name=%s icon=%s coverid=%s open_cmd=%s close_cmd=%s stop_cmd=%s", self, device, name, friendly_name, icon, coverid, open_cmd, close_cmd, stop_cmd)
+        super().__init__(device, config_entry, coverid, **kwargs)
         self._state = None
-        self._position = 50
+        self._position = None
+        self._current_cover_position = None
+        self._last_movement = None
+        self._last_position_set = None
+        self._last_command = None
+        #self._position = 50
+        _LOGGER.info("Initialized device=%s config_entry=%s coverid=%s", device, config_entry, coverid)
         print(
             "Initialized tuya cover  [{}] with switch status [{}] and state [{}]".format(
                 self.name, self._status, self._state
+
+    # def __init__(self, device, friendly_name, coverid, open_cmd, close_cmd, stop_cmd, set_position_dps, get_position_dps, last_movement_dps):
+    #     self._device = device
+    #     self._available = False 
+    #     self._name = friendly_name
+    #     self._cover_id = coverid
+    #     self._status = None
+    #     self._state = None
+    #     self._last_movement = None
+    #     self._last_position_set = None
+    #     self._last_command = None
+    #     self._current_cover_position = 50
+    #     self._open_cmd = open_cmd
+    #     self._close_cmd = close_cmd
+    #     self._stop_cmd = stop_cmd
+    #     self._get_position_dps = get_position_dps
+    #     self._set_position_dps = set_position_dps
+    #     self._last_movement_dps = last_movement_dps
+        
+    #     print(
+    #         "Initialized tuya cover [{}] with cover status [{}] and state [{}]".format(
+    #             self._name, self._status, self._state
             )
         )
 
+    @property
+    def device_info(self):
```

Comment:
You can remove `device_info` altogether as that is handled by `LocalTuyaEntity` now.